cookbook/core/extend_flyte/external_plugin_service.py

```diff
@@ -0,0 +1,119 @@
+  :alt: External-Plugin-System
+  :class: with-shadow
+
+Register a new plugin
+=====================
+
+Flytekit Interface Specification
+--------------------------------
+New backend plugins must be registered by extending the BackendPluginBase class in the flytekit backend module. All three methods below must be implemented, and all calls must be idempotent.
+
+- `create` is used to launch a new task. People can use gRPC, REST, or SDK to create a task.
+- `get` is used to get the jobID (like BigQuery Job ID or Databricks task ID).
+- `delete` Will send a request to delete the job.
+
+.. code-block:: python
+
+    class BackendPluginBase:
+    def __init__(self, task_type: str):
+        self._task_type = task_type
+
+    @property
+    def task_type(self) -> str:
+        return self._task_type
+
+    @abstractmethod
+    def create(
+        self,
+        context: grpc.ServicerContext,
+        output_prefix: str,
+        task_template: TaskTemplate,
+        inputs: typing.Optional[LiteralMap] = None,
+    ) -> TaskCreateResponse:
+        pass
+
+    @abstractmethod
+    def get(self, context: grpc.ServicerContext, job_id: str) -> TaskGetResponse:
+        pass
+
+    @abstractmethod
+    def delete(self, context: grpc.ServicerContext, job_id: str) -> TaskDeleteResponse:
+        pass
+
+    # To register the custom plugin
+    BackendPluginRegistry.register(CustomPlugin())
+
+Build a New image
+-----------------
+The following is a sample Dockerfile for building an image for an external plugin system.
+
+.. code-block:: Dockerfile
+
+    FROM python:3.9-slim-buster
+
+    MAINTAINER Flyte Team <users@flyte.org>
+    LABEL org.opencontainers.image.source=https://github.com/flyteorg/flytekit
+
+    WORKDIR /root
+    ENV PYTHONPATH /root
+
+    # flytekit will auto load the plugin if package is installed.
+    RUN pip install flytekitplugins-bigquery
+    CMD pyflyte serve --port 80
+
+Update Helm Chart
+-----------------
+1. Update the `images <https://github.com/flyteorg/flyte/pull/3454/files#diff-cbcadfa3abd9e4771deaab1b0f28d2c7f6a67f3ebd9b1b1ec6a0eebb38a718e4R142-R146>`__
+2. Update the FlytePropeller configmap.
+
+.. code-block:: YAML
+
+    tasks:
+      task-plugins:
+        enabled-plugins:
+          - external-plugin-service
+        default-for-task-types:
+          - bigquery_query_job_task: external-plugin-service
+          - custom_task: external-plugin-service
+
+    flyteplugins-service:
+      # By default, all the request will be sent to the default endpoint.
+      defaultGrpcEndpoint: "dns:///external-plugin-service-production.flyte.svc.cluster.local:80"
+      supportedTaskTypes:
+        - custom_task
+      endpointForTaskTypes:
+        # It will override the default grpc endpoint for bigquery endpoint, which means propeller will send create request to this endpoint.
+        - bigquery_query_job_task: "dns:///external-plugin-service-development.flyte.svc.cluster.local:80"
+
+3. Restart the FlytePropeller
```

Comment:
Can you include commands for restarting propeller and updating the propeller configmap?